src/Provider/ProductIndexScopeProviderInterface.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusAlgoliaPlugin\Provider;
+
+use Setono\SyliusAlgoliaPlugin\DTO\ProductIndexScope;
+
+interface ProductIndexScopeProviderInterface
+{
+    /**
+     * @return iterable<ProductIndexScope>
+     */
+    public function resolve(): iterable;
```

Comment:
This method name should be renamed also to `getProductIndexScopes` or something like that